packages/babel-generator/src/printer.js

```diff
@@ -484,8 +485,12 @@ export default class Printer {
   _printComment(comment) {
     if (!this._shouldPrintComment(comment)) return;
 
+    // Some plugins use this to mark comments as removed using the AST-root 'comments' property,
+    // where they can't manually mutate the AST node comment lists.
     if (comment.ignore) return;
-    comment.ignore = true;
+
+    if (this._printedComments.has(comment)) return;
+    this._printedComments.add(comment);
```

Comment:
So this is preventing a comment from being printed twice? was that an issue before?